FWCore/Utilities: remove intrusive macros breaking libstdc++

davidlt opened the pull request:
We cannot redefine 'private' and 'protected' keywords via macros to e.g.
'public'. This is extremely intrusive and breaks encapsulation.

This does not work anymore with new libstdc++ libraries, because foward
delcaration of struct is implicitly private and then implementation is
under explicit private clause. Redefining 'private' only change one of
them thus creating compile-time errors in sstream.

Details in PR65899 (GCC BZ). It's WONTFIX.

Such cleanups are required for GCC 5 and above.

Signed-off-by: David Abdurachmanov David.Abdurachmanov@cern.ch

cmsbuild commented:
A new Pull Request was created by @davidlt  for CMSSW_8_0_X.

FWCore/Utilities: remove intrusive macros breaking libstdc++

It involves the following packages:

FWCore/Utilities


@cmsbuild, @smuzaffar, @Dr15Jones can you please review it and eventually sign? Thanks.
@makortel, @Martin-Grunewald, @wddgit, @wmtan this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

cmsbuild commented:
+1
Tested at: 943be858c4065a28d4f49a352a2a5b62d69241c9
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12092/9188/summary.html

Dr15Jones commented on FWCore/Utilities/interface/RunningAverage.h:
```diff
@@ -4,10 +4,20 @@
 
+// Function for testing RunningAverage
+namespace test {
+  namespace RunningAverage {
```
CMS naming convention is namespaces start with a lower case letter.

davidlt replied:
Updated.

Commit:
FWCore/Utilities: remove intrusive macros breaking libstdc++

We cannot redefine 'private' and 'protected' keywords via macros to e.g.
'public'. This is extremely intrusive and breaks encapsulation.

This does not work anymore with new libstdc++ libraries, because foward
delcaration of struct is implicitly private and then implementation is
under explicit private clause. Redefining 'private' only change one of
them thus creating compile-time errors in sstream.

Details in PR65899 (GCC BZ). It's WONTFIX.

Such cleanups are required for GCC 5 and above.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Dr15Jones commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_0_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
Pull request #12092 was updated. @cmsbuild can you please check and sign again.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12092/9188/summary.html

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_0_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
+1
Tested at: 7a0a8ef2f3b4baea625953f20de4dd398962d693
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12092/9195/summary.html

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_0_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12092/9195/summary.html

davidlange6 commented:
+1
